eva/binder/statement_binder.py

```diff
@@ -151,7 +152,34 @@ def _bind_delete_statement(self, node: DeleteTableStatement):
     @bind.register(CreateMaterializedViewStatement)
     def _bind_create_mat_statement(self, node: CreateMaterializedViewStatement):
         self.bind(node.query)
-        # Todo Verify if the number projected columns matches table
+        num_projected_columns = 0
+        # TODO we should fix the binder. All binded object should have the same interface.
+        for expr in node.query.target_list:
+            if expr.etype == ExpressionType.TUPLE_VALUE:
+                num_projected_columns += 1
+            elif expr.etype == ExpressionType.FUNCTION_EXPRESSION:
+                num_projected_columns += len(expr.output_objs)
+            else:
+                raise BinderError("Unsupported expression type {}.".format(expr.etype))
+
+        assert (
+            len(node.col_list) == 0 or len(node.col_list) == num_projected_columns
+        ), "Projected columns mismatch, expected {} found {}".format(len(node.col_list), num_projected_columns)
+        binded_col_list = []
+        idx = 0
+        for expr in node.query.target_list:
+            output_objs = [(expr.col_name, expr.col_object)] if expr.etype == ExpressionType.TUPLE_VALUE else zip(expr.projection_columns, expr.output_objs)
```

Comment:
just made a commit here. `expr.col_object.name` does not always infer the correct column name (i.e. 'labels' != 'label'). `expr.col_name` fixed this.

Reply:
Thanks. Maybe I missed. What is the `expr.col_object.name` then?

Reply:
I'm not sure where it is set. It seems that the object's name attribute doesn't always match column name. For instance, the previous failed test had object name = "labels" but object column name = "label", and "label" should be the correct one passed all the way down to sql storage engine.

Reply:
@gaurav274 Is this a BUG or intended behavior?